backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2Native.kt

```diff
@@ -48,15 +49,20 @@ class K2Native : CLICompiler<K2NativeCompilerArguments>() {
             return ExitCode.OK
         }
 
-        if (arguments.freeArgs.isEmpty() && !arguments.isUsefulWithoutFreeArgs) {
-            configuration.report(ERROR, "You have not specified any compilation arguments. No output has been produced.")
-        }
-
         val environment = KotlinCoreEnvironment.createForProduction(rootDisposable,
             configuration, EnvironmentConfigFiles.NATIVE_CONFIG_FILES)
         val project = environment.project
         val konanConfig = KonanConfig(project, configuration)
 
+        if (arguments.freeArgs.isEmpty() && !arguments.isUsefulWithoutFreeArgs && !konanConfig.linkOnly) {
```

Comment:
Not really very readable, maybe somehow reorganize.

Reply:
Done